### ISSUE TITLE
train with multi gpus, always blocking...

### ISSUE THREAD

**Issue by mynameischaos** [open-mmlab/mmdetection#8265]
1. train with 2 nodes(a & b), each node has 8 gpus.
2. run with the following shell scripts：./tools/dist_train.sh configs/goods/faster_rcnn_r50_fpn_2x_coco.py 8

in dist_train.sh:
node a is :
#!/usr/bin/env bash

CONFIG=$1
GPUS=$2
NNODES=${NNODES:-2}
NODE_RANK=${NODE_RANK:-0}
PORT=${PORT:-29511}
MASTER_ADDR=${MASTER_ADDR:-"10.193.21.228"}

PYTHONPATH="$(dirname $0)/..":$PYTHONPATH \
python -m torch.distributed.launch \
    --nnodes=$NNODES \
    --node_rank=$NODE_RANK \
    --master_addr=$MASTER_ADDR \
    --nproc_per_node=$GPUS \
    --master_port=$PORT \
    $(dirname "$0")/train.py \
    $CONFIG \
    --seed 0 \
    --launcher pytorch ${@:3}

node b is:

#!/usr/bin/env bash

CONFIG=$1
GPUS=$2
NNODES=${NNODES:-2}
NODE_RANK=${NODE_RANK:-1}
PORT=${PORT:-29511}
MASTER_ADDR=${MASTER_ADDR:-"10.193.21.228"}

PYTHONPATH="$(dirname $0)/..":$PYTHONPATH \
python -m torch.distributed.launch \
    --nnodes=$NNODES \
    --node_rank=$NODE_RANK \
    --master_addr=$MASTER_ADDR \
    --nproc_per_node=$GPUS \
    --master_port=$PORT \
    $(dirname "$0")/train.py \
    $CONFIG \
    --seed 0 \
    --launcher pytorch ${@:3}

But, the programming will block and print ：
/mnt/longvideo/zhonghuasong/anaconda3/envs/openmmlab/lib/python3.8/site-packages/torch/distributed/launch.py:178: FutureWarning: The module torch.distributed.launch is deprecated
and will be removed in future. Use torchrun.
Note that --use_env is set by default in torchrun.
If your script expects `--local_rank` argument to be set, please
change it to read from `os.environ['LOCAL_RANK']` instead. See
https://pytorch.org/docs/stable/distributed.html#launch-utility for
further instructions

  warnings.warn(
WARNING:torch.distributed.run:
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed.
*****************************************

Please help me.

**Comment by chhluo** [comment]
The command you used is right. 

Please check whether the nodes can reach each other and whether `NCCL_SOCKET_IFNAME` is set correctly (set to the interface which has been bound with ip)  , and add `export NCCL_DEBUG=INFO` before runing.

**Comment by mynameischaos** [comment]
> The command you used is right.
> 
> Please check whether the nodes can reach each other and whether `NCCL_SOCKET_IFNAME` is set correctly (set to the interface which has been bound with ip) , and add `export NCCL_DEBUG=INFO` before runing.
-----------------------------------
The nodes can ping each other. And export NCCL_SOCKET_IFNAME=eth0 and export NCCL_DEBUG=INFO. Then run again. but still blocking...

**Comment by chhluo** [comment]
> > The command you used is right.
> > Please check whether the nodes can reach each other and whether `NCCL_SOCKET_IFNAME` is set correctly (set to the interface which has been bound with ip) , and add `export NCCL_DEBUG=INFO` before runing.
> 
> The nodes can ping each other. And export NCCL_SOCKET_IFNAME=eth0 and export NCCL_DEBUG=INFO. Then run again. but still blocking...

`NCCL_SOCKET_IFNAME` may not be `eth0` when there two or more network cards on one node.
